src/halmos/__main__.py

```diff
@@ -730,7 +626,46 @@ def run_target_contract(ctx: ContractContext, ex: Exec, addr: Address) -> list[E
         finally:
             reset(solver)
 
-    return results
+
+def get_frontier(ctx: ContractContext, depth: int) -> Iterable[Exec]:
+    """
+    Retrieves the frontier states at a given depth.
+
+    If the frontier states have already been computed, the cached results are returned.
+    Otherwise, the generator from compute_frontier() is returned.
+    """
```

Comment:
makes sense, I would add a note that this is not thread safe, it is meant to be only called from a single thread. Otherwise we could potentially get things like an incomplete `frontier` (while `compute_frontier` is still running)

Reply:
https://github.com/a16z/halmos/pull/472/commits/7b9503493e36795034fd37cfce6423dfbd9bd7ba